Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs

```diff
@@ -281,6 +289,18 @@ public Task<BoolResult> RequestCopyFileAsync(OperationContext context, ContentHa
             {
                 var location = hashInfo.Locations[replicaIndex];
 
+                //Currently everytime we increment attemptCount's value, we go through every location in hashInfo and try to copy.
+                //We add one because replicaIndex is indexed from zero.
+                //If we reach over maximum retries, return an put result stating so, and no longer retry
+                if ((attemptCount * hashInfo.Locations.Count + replicaIndex + 1) > MaxRetryCount)
+                {
+                    Tracer.Debug(
+                            context,
+                            $"{AttemptTracePrefix(attemptCount)} Reached maximum number of total retries.");

```

Comment:
I think that both - the tracing and the resulting error message should contain the number of attempts that we tried. I.e. MaxRetryCount.